### PR TITLE
Use SecretManager when getting secrets for EnvFrom

### PR DESCRIPTION
**Pull request by fraenkel**
Merges crossed in the night which missed this needed change.

### CODE REVIEW

**Comment by fraenkel** [comment]
@pmorie The one that got away....

**Comment by resouer** [comment]
@k8s-bot ok to test

**Comment by fraenkel** [comment]
@pmorie PTAL

**Comment by fraenkel** [comment]
@liggitt rebased but shouldn't be an issue. It was an issue for my other one regarding optional secrets which I just rebased.

File: pkg/kubelet/kubelet_test.go [hunk 1]
```diff
@@ -173,8 +173,12 @@ func newTestKubeletWithImageList(
 	kubelet.cadvisor = mockCadvisor
 
 	fakeMirrorClient := podtest.NewFakeMirrorClient()
-	fakeSecretManager := secret.NewFakeManager()
```
Comment (liggitt) [inline pkg/kubelet/kubelet_test.go]: Why not a fake one?

Comment (fraenkel) [inline pkg/kubelet/kubelet_test.go]: Look at what it does. It doesn't work at all in any real case that will request a secret.
By using the real one, my tests now pass. Just ran into this in the optional configmap and secrets too.

**Comment by Random-Liu** [comment]
@yujuhong Do you want to take a look at this PR? It seems that you reviewed the initial PR. :)

**Comment by yujuhong** [comment]
> @yujuhong Do you want to take a look at this PR? It seems that you reviewed the initial PR. :)

Sure.

/cc @wojtek-t

**Comment by wojtek-t** [comment]
I took a look and this LGTM. But I will leave the final review for @yujuhong

**Pull request merged**